src/main/java/com/exactpro/th2/common/schema/event/EventBatchSender.java

```diff
@@ -40,17 +42,26 @@ public EventBatchSender(
             @NotNull ConnectionManager connectionManager,
             @NotNull String exchangeName,
             @NotNull String routingKey,
-            @NotNull String th2Pin
+            @NotNull String th2Pin,
+            @NotNull String bookName
     ) {
-        super(connectionManager, exchangeName, routingKey, th2Pin, EVENT_TYPE);
+        super(connectionManager, exchangeName, routingKey, th2Pin, EVENT_TYPE, bookName);
     }
 
     @Override
     public void send(EventBatch value) throws IOException {
         EVENT_PUBLISH_TOTAL
                 .labels(th2Pin)
                 .inc(value.getEventsCount());
-        super.send(value);
+        EventBatch.Builder eventBatchBuilder = EventBatch.newBuilder();
```

Comment:
Execute a lightweight check before rebuilding the batch to avoid redundant operations.

Reply:
done